src/Microsoft.Health.Fhir.Core/Features/Search/SearchService.cs

```diff
@@ -72,21 +75,59 @@ public async Task<Bundle> SearchCompartmentAsync(string compartmentType, string
             string resourceId,
             PartialDateTime at,
             PartialDateTime since,
+            PartialDateTime before,
             int? count,
             string continuationToken,
             CancellationToken cancellationToken)
         {
             var queryParameters = new List<Tuple<string, string>>();
+            DateTimeOffset? addedBefore = null;
 
-            if (at != null && since != null)
+            if (at != null)
             {
-                // _at and _since cannot be both specified.
-                throw new InvalidSearchOperationException(
-                    string.Format(
-                        CultureInfo.InvariantCulture,
-                        Core.Resources.AtAndSinceCannotBeBothSpecified,
-                        KnownQueryParameterNames.At,
-                        KnownQueryParameterNames.Since));
+                if (since != null)
+                {
+                    // _at and _since cannot be both specified.
+                    throw new InvalidSearchOperationException(
+                        string.Format(
+                            CultureInfo.InvariantCulture,
+                            Core.Resources.AtAndSinceCannotBeBothSpecified,
+                            KnownQueryParameterNames.At,
+                            KnownQueryParameterNames.Since));
+                }
+
+                if (before != null)
+                {
+                    // _at and _since cannot be both specified.
+                    throw new InvalidSearchOperationException(
+                        string.Format(
+                            CultureInfo.InvariantCulture,
+                            Core.Resources.AtAndSinceCannotBeBothSpecified,
```

Comment:
nit: should rename this string since it's not really specifically for at and since.